tests/baselines/reference/mergeTwoInterfaces3.types

```diff
@@ -0,0 +1,106 @@
+>(p: string) => null! : (p: string) => any
+>p : string
+>null! : null
+
+m((p: string, e: string) => null!);
+>m((p: string, e: string) => null!) : Chainable<string, string>
+>m : { (fn: (prevSubject: string) => Chainable<string, undefined>): Chainable<string, undefined>; (fn: (prevSubject: string, extra: string) => Chainable<string, undefined>): Chainable<string, string>; }
+>(p: string, e: string) => null! : (p: string, e: string) => any
+>p : string
+>e : string
+>null! : null
+
+m((p) => null!);
+>m((p) => null!) : Chainable<string, string>
+>m : { (fn: (prevSubject: string) => Chainable<string, undefined>): Chainable<string, undefined>; (fn: (prevSubject: string, extra: string) => Chainable<string, undefined>): Chainable<string, string>; }
+>(p) => null! : (p: string) => any
+>p : string
+>null! : null
+
+m((p, e) => null!);
+>m((p, e) => null!) : Chainable<string, string>
+>m : { (fn: (prevSubject: string) => Chainable<string, undefined>): Chainable<string, undefined>; (fn: (prevSubject: string, extra: string) => Chainable<string, undefined>): Chainable<string, string>; }
+>(p, e) => null! : (p: string, e: string) => any
+>p : string
+>e : string
+>null! : null
+
+declare let o: Ordinary;
+>o : { (fn: (prevSubject: string) => Chainable<string, undefined>): Chainable<string, undefined>; (fn: (prevSubject: string, extra: string) => Chainable<string, undefined>): Chainable<string, string>; }
+
+o((p: string) => null!);
+>o((p: string) => null!) : Chainable<string, undefined>
+>o : { (fn: (prevSubject: string) => Chainable<string, undefined>): Chainable<string, undefined>; (fn: (prevSubject: string, extra: string) => Chainable<string, undefined>): Chainable<string, string>; }
+>(p: string) => null! : (p: string) => any
+>p : string
+>null! : null
+
+o((p: string, e: string) => null!);
+>o((p: string, e: string) => null!) : Chainable<string, string>
+>o : { (fn: (prevSubject: string) => Chainable<string, undefined>): Chainable<string, undefined>; (fn: (prevSubject: string, extra: string) => Chainable<string, undefined>): Chainable<string, string>; }
+>(p: string, e: string) => null! : (p: string, e: string) => any
+>p : string
+>e : string
+>null! : null
+
+o((p) => null!);
+>o((p) => null!) : Chainable<string, undefined>
+>o : { (fn: (prevSubject: string) => Chainable<string, undefined>): Chainable<string, undefined>; (fn: (prevSubject: string, extra: string) => Chainable<string, undefined>): Chainable<string, string>; }
+>(p) => null! : (p: string) => any
+>p : string
+>null! : null
+
+o((p, e) => null!);
+>o((p, e) => null!) : Chainable<string, string>
+>o : { (fn: (prevSubject: string) => Chainable<string, undefined>): Chainable<string, undefined>; (fn: (prevSubject: string, extra: string) => Chainable<string, undefined>): Chainable<string, string>; }
+>(p, e) => null! : (p: any, e: any) => any

```

Comment:
To me, it seems like there may be a bug. The type of both `p` and `e` should have been inferred as `string`.

https://www.typescriptlang.org/play?noUncheckedIndexedAccess=true&exactOptionalPropertyTypes=true&ts=5.0.4#code/JYOwLgpgTgZghgYwgAgMIAs6jgIwDYQA8AygK44BWECYyAvMnCAJ4B8yA3gLABQy-yAO7Aw6UAAoYIAFzJxABygQAbmUrUwstVRoBKeuwxYQuAiXI6wrXbKPZ8RbRtYBuXgF9evUJFiIUdiYO5uo09IwsADTIAKIAHmBQcAAqzPIoDKQgACYQMKAQ2ezcfALCohJSsgpKqhYaWvU00RAJSbLxiSlpEPp0hpj2Zk401raDQcNNYNGdSanprh5ePD7Q8EhoE6YQADLAANaO0+FMzLNt3enhWbn5IIXFvAJCImIgkjJyiiojmsh-PoDYw7EKWMZbEHBP5LUr8crvT7VH51UL-P4tS4dS4LXoGSFDY5oiGBUEY2I4nqwzw8XhgHoEyYQADqb2I1AA9jlceFScEAM6JUAAc2igqgIth9OuAFloMLCrztg5WaJ2QgudlcQBtADkCNAuoAum4eNKUAB5KDZbBQZhKqEEfZHQjikVioUgYWsPUGkDG028XIIPBwJTIAi0AC2sjlUAV2VNUfEClkbq9QOQIFIeDwAEJdEmU-I057RcgIKWJRn8dncwWiwpM3X84XeMmFC1mznW4GeMHQ+HI8gObIrTaTHbTRzi1WRd3622eDPU8h0+XK2uywve7wV-Idw298Wu7Wew2gA